internal/sdkv2provider/resource_cloudflare_workers_queue.go

```diff
@@ -0,0 +1,131 @@
+package sdkv2provider
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/cloudflare/cloudflare-go"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/pkg/errors"
+)
+
+func resourceCloudflareWorkersQueue() *schema.Resource {
+	return &schema.Resource{
+		Schema:        resourceCloudflareWorkersQueueSchema(),
+		CreateContext: resourceCloudflareWorkersQueueCreate,
+		ReadContext:   resourceCloudflareWorkersQueueRead,
+		UpdateContext: resourceCloudflareWorkersQueueUpdate,
+		DeleteContext: resourceCloudflareWorkersQueueDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceCloudflareWorkersQueueImport,
+		},
+		Description: "Provides the ability to manage Cloudflare Workers Queue features.",
+	}
+}
+
+func resourceCloudflareWorkersQueueCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+
+	accountID := d.Get("account_id").(string)
+	if accountID == "" {
+		accountID = client.AccountID
+	}
+
+	queueName := d.Get("name").(string)
+	req := cloudflare.CreateQueueParams{
+		Name: queueName,
+	}
+
+	tflog.Debug(ctx, fmt.Sprintf("[Info] Creating Cloudflare Workers Queue from struct: %+v", req))
+
+	r, err := client.CreateQueue(ctx, cloudflare.AccountIdentifier(accountID), req)
+	if err != nil {
+		return diag.FromErr(errors.Wrap(err, "error creating workers queue"))
+	}
+
+	if r.ID == "" {
+		return diag.FromErr(fmt.Errorf("failed to find id in Create response; resource was empty"))
+	}
+
+	d.SetId(r.ID)
+
+	tflog.Info(ctx, fmt.Sprintf("Cloudflare Workers Queue ID: %s. Name: %s", d.Id(), queueName))
+
+	return resourceCloudflareWorkersQueueRead(ctx, d, meta)
+}
+
+func resourceCloudflareWorkersQueueRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*cloudflare.API)
+	queueID := d.Id()
+
+	accountID := d.Get("account_id").(string)
+	if accountID == "" {
+		accountID = client.AccountID
```

Comment:
similar to above, this should be removed in favour of the explicit value.

Reply:
✅